lib/ably/realtime/connection.rb

```diff
@@ -615,6 +615,8 @@ def connection_resumable?
       def connection_state_available?
         return true if connected?
 
+        return false if time_since_connection_confirmed_alive? > connection_state_ttl
+
         connected_last = state_history.reverse.find { |connected| connected.fetch(:state) == :connected }
         if connected_last.nil? || (connected_last.fetch(:transitioned_at) < Time.now - connection_state_ttl)
           false
```

Comment:
I know not part of this PR, but that doesn't look right -- you don't resume if the most recent time you transitioned to­ (not from) `connected` was more than connectionStateTtl ago? Am I reading that correctly?

Reply:
Yes, you are reading it incorrectly, it's time since connection comfirmed alive i.e. freshness.  This existing method is used to detect a connection that is no longer active.

Reply:
> Yes, you are reading it incorrectly, it's time since connection comfirmed alive i.e. freshness. 

No, that comment is on lines 619-620 (hence 'I know not part of this PR'). This bit: `if connected_last.nil? || (connected_last.fetch(:transitioned_at) < Time.now - connection_state_ttl)`

> This existing method is used to detect a connection that is no longer active.

Yes, I can see that from the method name, I'm just not sure the logic is correct.

Reply:
To be explicit: if I'm reading this correctly, line 621 says "don't resume if the time of the last transition to `connected` is over 2 minutes ago". So if you're connected for an hour, then your connection drops for 5 seconds, it won't resume because the last transition to `connected` was an hour ago.

Reply:
Mmm, I didn't look at that code, but that does look suspect. I will investigate.

Reply:
@SimonWoolf btw. I pushed b70811e earlier, will see what that breaks :)